src/platform_impl/linux/wayland/window.rs

```diff
@@ -441,6 +441,8 @@ impl WindowStore {
             bool,
             bool,
             bool,
+            Option<bool>,
+            &wl_surface::WlSurface,
```

Comment:
Is there any particular reason we don't just pass a reference to the entire internal window instead of many bit and pieces of it?

Reply:
They wont be the same and we can't really pass it, since we want to mutate existing one and send something different (`replace`, `take`). `make_wid` doesn't make sense in event loop proxy either. We can try to do it ofc, but I'd like to leave it as it is.

@vberger probably have some explanation on why things are written this way though.

Reply:
To be quite honest I don't remember exactly the reason I set it like that originally, most likely I didn't find a better way. And then things grew on top of it without changing the pattern.

There may certainly be a better way to structure this, but I agree this would be for an other PR. In general I suspect there is room for some refactoring / reorganization of winit's wayland backend, and I also suspect this would better be done in link with https://github.com/Smithay/client-toolkit/issues/77.